### PR TITLE
Add functionality so that if one entry breaks during a batch, it will…

### PR DESCRIPTION
**Pull request by jamesuejio**
… not crash the entire batch

### CODE REVIEW

File: ach/builder.py [hunk 1]
```diff
@@ -79,35 +79,38 @@ def add_batch(self, std_ent_cls_code, batch_entries=None,
...
+                entry.amount = int(round(float(record['amount']) * 100))
+                entry.ind_name = record['name'].upper()[:22]
+                entry.trace_num = self.settings['immediate_dest'][:8] \
+                    + entry.validate_numeric_field(entry_counter, 7)
 
-            entries.append((entry, record.get('addenda', [])))
-            entry_counter += 1
+                entries.append((entry, record.get('addenda', [])))
+                entry_counter += 1
+            except:
```
Comment (felippemr) [inline ach/builder.py]: Can we return what type of failure it was? 

Swallowing the error on a bunch of lines will help but, I believe we can do a better job by informing the user what the errors are or where they happened.

**Review by felippemr — APPROVED** [review]

File: ach/builder.py (hunk 1 above)
Comment (jamesuejio) [reply ach/builder.py]: Sure how do you think we should return the error? Should I add the error message in the entry? Or have failed_entries a tuple of entry dict and error message?

Comment (jayneflux) [reply ach/builder.py]: I like the tuple idea.

Comment (felippemr) [reply ach/builder.py]: Tuple is ok for now but we could return an either type in the future.

**Comment by natalya-carta** [comment]
Changes look awesome, I feel like there should be an obvious test we can add?

**Comment by jamesuejio** [comment]
There are tests in https://github.com/carta/carta-payments/pull/931 I think writing tests for this library is out of scope.

**Comment by natalya-carta** [comment]
> There are tests in [carta/carta-payments#931](https://github.com/carta/carta-payments/pull/931) I think writing tests for this library is out of scope.

SG, LGTM'd there.

**Review by natalya-carta — APPROVED** [review]
LGTM

File: ach/builder.py (hunk 1 above)
Comment (natalya-carta) [reply ach/builder.py]: Hmm, Felippe, looking closer at your comment you're totally right. Can we log it with some kind of unique identifier to find that detail if not obvious from the data itself returned to the caller? This is a library so not sure what the convention is in python for logging in libraries, thoughts?

**Comment by jamesuejio** [comment]
Oh there are tests in this repo! Let me add a quick one

**Comment by jamesuejio** [comment]
Actually can't figure out how to run them and the last commits to tests were 6 years so not sure it's worth trying to figure it out.

**Comment by natalya-carta** [comment]
> Actually can't figure out how to run them and the last commits to tests were 6 years so not sure it's worth trying to figure it out.

From my quick look, it's `travis` based and there are ways to run it locally (e.g. https://stackoverflow.com/questions/21053657/how-to-run-travis-ci-locally). I think we should either delete them or convert to something our CI can run - my strong preference is for the latter, I like the fact that the tests give a quick and easy idea of what the file should look like.

**Comment by jamesuejio** [comment]
@natalya-carta thank you for coming back and commenting. This repo is actually public so not sure if we can configure so our CI can run. Will task.

**Comment by asinha10** [comment]
Hi,

I was using this library to generate a NACHA CCD payment file based on the entries given in the example on pypi. I found one problem with library. 
Suppose the company name is "ABC INC (USA)", the brackets '(' are not recognized and the entire line is blank after the "ABC INC".
Similarly, if Addenda records has * in the text like `0705RMR*IV*234827`, the line breaks after 0705RMR.
Can you help with this issue specially with * in addenda `RMR*IV*`, because CCD format should have `RMR*IV*` for payment remittance.

Thank You.

**Pull request merged**